packages/next/server/web-server.ts

```diff
@@ -175,7 +174,9 @@ export default class NextWebServer extends BaseServer {
         // Not implemented: on/removeListener
       } as any)
     } else {
-      res.body(await options.result.toUnchunkedString())
+      // TODO: generate Etag
+      const payload = await options.result.toUnchunkedString()
+      res.body(payload)
```

Comment:
Yep we need a lighter way to generate etag in this runtime, based on Web Crypto APIs.